x-pack/qa/third-party/active-directory/src/test/java/org/elasticsearch/xpack/security/authc/ldap/AbstractAdLdapRealmTestCase.java

```diff
@@ -265,12 +275,13 @@ protected void assertAccessAllowed(String user, String index) throws IOException
     }
 
     protected void assertAccessDenied(String user, String index) throws IOException {
+        final Client client = client().filterWithHeader(Collections.singletonMap(BASIC_AUTH_HEADER, userHeader(user, PASSWORD)));
+        // Force an authentication to populate the cache.
+        // We can safely re-try this if it fails, which means we can be more confident that the index request failed for the correct reason
+        authenticateUser(client, user, 3);
```

Comment:
What do you say if we move this bits in a method like `assertAdFixtureIsActive` (maybe followed by a realm cache call)  and call that in a `@Before` method.
I think this would scoped tighter to the lazy fixture problem. Scatering authenticate calls before every index runs the slight potential to hide true issues (and it makes test console dumps harder to follow).

Reply:
The tricky bit is that we need to do it for each user - the problem is that we need every user to be in the cache so we avoid external LDAP calls during authz checks

That could be done, but it's not just one user - it's several and we'd need to pre-cache each of them.
https://github.com/elastic/elasticsearch/blob/a92a647b9f17d1bddf5c707490a19482c273eda3/x-pack/qa/third-party/active-directory/src/test/java/org/elasticsearch/xpack/security/authc/ldap/GroupMappingIT.java#L23-L27

I'm not sure that ends up being a nicer solution.